public_html/wp-content/plugins/camptix-attendance/addons/attendance.php

```diff
@@ -270,16 +309,38 @@ public function _filter_query_search( $search ) {
 				INNER JOIN $wpdb->postmeta tix_last_name ON ( ID = tix_last_name.post_id AND tix_last_name.meta_key = 'tix_last_name' )
 			";
 
-			$clauses['where'] .= $wpdb->prepare( "
-				AND (
+				$where = $wpdb->prepare(
+					"
+				(
 					tix_first_name.meta_value LIKE '%%%s%%' OR
 					tix_last_name.meta_value LIKE '%%%s%%' OR
 					CONCAT( tix_first_name.meta_value, ' ', tix_last_name.meta_value ) LIKE '%%%s%%'
 				)
-			", $search, $search, $search );
-
-			return $clauses;
-		} );
+			",
+					$search,
+					$search,
+					$search
+				);
+
+				// Match against the Access Token, specifically the second half of the hash, if it looks like one.
+				if ( preg_match( '!^[0-9a-f]{16}$!i', $search ) ) {
+					$clauses['join'] .= "
+					INNER JOIN $wpdb->postmeta tix_access_token ON ( ID = tix_access_token.post_id AND tix_access_token.meta_key = 'tix_access_token' )
+				";
+
+					$where = '(' .
+						$where .
+						' OR (' .
+						$wpdb->prepare( "tix_access_token.meta_value LIKE '%%%s'", $search ) .
+						')' .
+					')';
+				}
+
+				$clauses['where'] .= ' AND ' . $where;
```

Comment:
Maybe put the previous clauses into two variables and "join" them differently, then overwriting `$where` multiple times.

Reply:
we could do an array an join them later? Skip this for now